src/test/scala/io/qbeast/spark/utils/QbeastOptimizeIntegrationTest.scala

```diff
@@ -107,4 +109,182 @@ class QbeastOptimizeIntegrationTest extends QbeastIntegrationTestSpec {
       filesAfter.map(_.blocks.size).sum should be < filesBefore.map(_.blocks.size).sum
   }
 
+  it should "optimize a table converted to Qbeast" in withQbeastContextSparkAndTmpDir {
+    (spark, tmpDir) =>
+      spark
+        .range(50)
+        .write
+        .mode("append")
+        .format("delta")
+        .save(tmpDir) // Append data without indexing
+
+      ConvertToQbeastCommand(identifier = s"delta.`$tmpDir`", columnsToIndex = Seq("id"))
+        .run(spark)
+
+      val deltaLog = DeltaLog.forTable(spark, tmpDir)
+      val firstSnapshot = deltaLog.update()
+      val firstUnindexedFiles = firstSnapshot.allFiles.where("tags is null")
+      firstUnindexedFiles.count() shouldBe firstSnapshot.allFiles.count()
+      // Optimize the Table
+      val qt = QbeastTable.forPath(spark, tmpDir)
+      qt.optimize(0L)
+
+      // After optimization, all files from the Legacy Table should be indexed
+      val snapshot = deltaLog.update()
+      val unindexedFiles = snapshot.allFiles.where("tags is null") // no tags
+      unindexedFiles shouldBe empty
+  }
+
+  it should "Optimize and Hybrid Table" in withQbeastContextSparkAndTmpDir { (spark, tmpDir) =>
+    spark
+      .range(50)
+      .write
+      .mode("append")
+      .option("columnsToIndex", "id")
+      .format("qbeast")
+      .save(tmpDir)
+
+    spark
+      .range(50)
+      .write
+      .mode("append")
+      .format("delta")
+      .save(tmpDir) // Append data without indexing
+
+    val deltaLog = DeltaLog.forTable(spark, tmpDir)
+    val firstSnapshot = deltaLog.update()
+    val firstUnindexedFiles = firstSnapshot.allFiles.where("tags is null")
+    firstUnindexedFiles should not be empty
+
+    // Optimize the Table
+    val qt = QbeastTable.forPath(spark, tmpDir)
+    qt.optimize(0L)
+
+    // After optimization, all files from the Hybrid Table should be indexed
+    val snapshot = deltaLog.update()
+    val unindexedFiles = snapshot.allFiles.where("tags is null") // no tags
+    unindexedFiles shouldBe empty
+  }
+
+  it should "optimize a table with Deletes" in withQbeastContextSparkAndTmpDir {
```

Comment:
When using Delete and Update with Copy on Write, it will create new files with no tags. We should re-index them as part of the unindexed files, right?